### PR TITLE
Add delay on server

### PR DESCRIPTION
**Pull request by WerWebWer**
A new feature has been added. It allows you to create a delay on the server in milliseconds. You can use "--delay <unsigned int>"

### CODE REVIEW

File: .project [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (cail) [inline .project]: remove this file

File: src/iperf_server_api.c [hunk 2]
```diff
@@ -88,8 +88,11 @@ iperf_server_listen(struct iperf_test *test)
 
     if (!test->json_output) {
 	iperf_printf(test, "-----------------------------------------------------------\n");
-	iperf_printf(test, "Server listening on %d\n", test->server_port);
+	iperf_printf(test, "Server listening on %d", test->server_port);
+    if (test->delay > 0 ) iperf_printf(test, " with max delay = %i\n", test->delay);
```
Comment (cail) [inline src/iperf_server_api.c]: fix code style, each statement on separate line.

File: src/iperf_server_api.c [hunk 3]
```diff
@@ -469,63 +483,63 @@ iperf_run_server(struct iperf_test *test)
...
     
                     if ((s = test->protocol->accept(test)) < 0) {
-			cleanup_server(test);
+			            cleanup_server(test);
                         return -1;
-		    }
+		            }
 
 #if defined(HAVE_TCP_CONGESTION)
-		    if (test->protocol->id == Ptcp) {
```
Comment (cail) [inline src/iperf_server_api.c]: Why spacing only changes? revert these

File: src/iperf_server_api.c [hunk 4]
```diff
@@ -469,63 +483,63 @@ iperf_run_server(struct iperf_test *test)
...
+		        }
                 FD_CLR(test->ctrl_sck, &read_set);                
             }
 
             if (test->state == CREATE_STREAMS) {
                 if (FD_ISSET(test->prot_listener, &read_set)) {
     
                     if ((s = test->protocol->accept(test)) < 0) {
-			cleanup_server(test);
+			            cleanup_server(test);
```
Comment (cail) [inline src/iperf_server_api.c]: revert spacing changes

File: src/iperf_server_api.c [hunk 5]
```diff
@@ -469,63 +483,63 @@ iperf_run_server(struct iperf_test *test)
...
                         return -1;
-		    }
```
Comment (cail) [inline src/iperf_server_api.c]: revert spacing changes

File: src/iperf_server_api.c [hunk 6]
```diff
@@ -553,11 +567,11 @@ iperf_run_server(struct iperf_test *test)
                             if (s > test->max_fd) test->max_fd = s;
 
                             /*
-                             * If the protocol isn't UDP, or even if it is but
```
Comment (cail) [inline src/iperf_server_api.c]: revert spacing changes

File: .project (hunk 1 above)
Comment (WerWebWer) [reply .project]: I deleted this file

File: src/iperf_server_api.c [hunk 7]
```diff
@@ -433,23 +437,33 @@ iperf_run_server(struct iperf_test *test)
...
 
-	iperf_time_now(&now);
-	timeout = tmr_timeout(&now);
+	    iperf_time_now(&now);
+	    timeout = tmr_timeout(&now);
         result = select(test->max_fd + 1, &read_set, &write_set, NULL, timeout);
+        
+        // Delay
+        if ((test->delay > 0) && (result == 0)){
+            iperf_printf(test, "Delay start -> ");
```
Comment (cail) [inline src/iperf_server_api.c]: There will be a huge printout flood. Remove printf at all.

File: src/iperf_server_api.c [hunk 8]
```diff
@@ -433,23 +437,33 @@ iperf_run_server(struct iperf_test *test)
...
+            srand(time(NULL));
+            unsigned int i =rand()%(test->delay);
+            usleep(i*1000);
```
Comment (cail) [inline src/iperf_server_api.c]: How do you guarantee this delay will happen once every second?

**Review by cail — CHANGES_REQUESTED** [review]
general: revert spacing changes, see other comments as well.

**Comment by WerWebWer** [comment]
I reverted spacing changes. And delay will appear when the result is 0, and it depends on time

File: src/iperf_server_api.c [hunk 9]
```diff
@@ -88,7 +88,8 @@ iperf_server_listen(struct iperf_test *test)
 
     if (!test->json_output) {
 	iperf_printf(test, "-----------------------------------------------------------\n");
-	iperf_printf(test, "Server listening on %d\n", test->server_port);
+	if (test->delay > 0 ) iperf_printf(test, "Server listening on %d  with max delay = %i\n", test->server_port, test->delay);
```
Comment (cail) [inline src/iperf_server_api.c]: Separate `if` and statement on different lines: that a generic coding rule:

```
if (test->delay > 0 )
    iperf_printf(....)
else
    iperf_printf()
```

**Comment by WerWebWer** [comment]
With corrected if

**Comment by cail** [comment]
@WerWebWer plz fix the outstanding comments so we can merge in.

**Pull request merged**